src/main/java/com/iabtcf/decoder/ConsentParser.java

```diff
@@ -0,0 +1,311 @@
+     * defined in the Policies (Appendix A section E). A CMP that services multiple publishers sets
+     * this value to false.
+     *
+     * @since 2.0
+     * @return true if if the CMP used non-IAB standard stacks during consent gathering; false
+     *         otherwise.
+     */
+    boolean getUseNonStandardStacks();
+
+    /**
+     * The TCF Policies designates certain Features as "special" which means a CMP must afford the
+     * user a means to opt in to their use. These "Special Features" are published and numerically
+     * identified in the Global Vendor List separately from normal Features.
+     *
+     * @since 2.0
+     * @return the Special Features the Vendor may utilize when performing some declared Purposes
+     *         processing.
+     */
+    IntIterable getSpecialFeatureOptIns();
+
+    /**
+     * The Purpose’s transparency requirements are met for each Purpose on the legal basis of
+     * legitimate interest and the user has not exercised their "Right to Object" to that Purpose.
+     * 
+     * By default or if the user has exercised their "Right to Object" to a Purpose, the
+     * corresponding identifier for that Purpose is set to false.
+     * 
+     * @since 2.0
+     * @return The purpose identifiers for which the legal basis of legitimate interest are
+     *         established.
+     */
+    IntIterable getPurposesLITransparency();
+
+    /**
+     * CMPs can use the PublisherCC field to indicate the legal jurisdiction the publisher is under
+     * to help vendors determine whether the vendor needs consent for Purpose 1.
+     * 
+     * In a globally-scoped TC string, this field must always have a value of false. When a CMP
+     * encounters a globally-scoped TC String with PurposeOneTreatment set to true then it is
+     * considered invalid and the CMP must discard it and re-establish transparency and consent.
+     *
+     * @since 2.0
+     * @return true if Purpose 1 was NOT disclosed; false otherwise.
+     */
+    boolean getPurposeOneTreatment();
+
+    /**
+     * The country code of the country that determines legislation of reference. Commonly, this
+     * corresponds to the country in which the publisher’s business entity is established.
+     *
+     * @since 2.0
+     * @return ISO 3166-1 alpha-2 country code
+     */
+    String getPublisherCC();
+
+    /**
+     * If a user exercises their "Right To Object" to a vendor’s processing based on a legitimate
+     * interest.
+     *
+     * @since 2.0
+     * @return vendor identifiers that can process this user based on legitimate interest
+     */
+    IntIterable getVendorLegitimateInterest();
+
+    /**
+     * The restrictions of a vendor's data processing by a publisher within the context of the users
+     * trafficking their digital property.
+     * 
+     * @since 2.0
+     * @return the list of publisher restrictions.
+     */
+    List<PublisherRestriction> getPublisherRestrictions();
```

Comment:
We need to stop returning a list here. I've made this point before that the list means nothing to the user and just requires them to iterate over the entire list to find the purposes they are consented to. We should either return a map keyed by purpose or vendor, or return a restriction type based on a vendor id and purpose id

Reply:
I haven't yet addressed PublisherRestrictions. Lets create an issue and discuss how we want to return this to users?